### PR TITLE
fix(HR): Appraisal form Enhancements

### PR DESCRIPTION
**Pull request by abhishekbalam**
The "Calculate Total Score" button does not look good and 
it feels obtrusive. An alternate solution without using the button.
The calculation is triggered by the child table field change events.

Before:
image

After:

[GIF Link](http://frappe-gifs.surge.sh/gifs/appraisal.gif)

Also made some edge case fixes where the Total Score was being set to NaN and improved a validation error message.

### CODE REVIEW

File: erpnext/hr/doctype/appraisal/appraisal.js [hunk 1]
```diff
@@ -10,45 +10,75 @@ frappe.ui.form.on('Appraisal', {
...
-			d.score_earned = 0;
+		if (flt(d.score) > 5) {
+			frappe.msgprint(__("Score must be less than or equal to 5"));
+			d.score = 0;
+			refresh_field('score', d.name, 'goals');
+		}
+		else {
+			frm.trigger('set_score_earned');
 		}
+<<<<<<< Updated upstream
```
Comment (Anurag810) [inline erpnext/hr/doctype/appraisal/appraisal.js]: Please resolve conflicts

**Review by Anurag810 — CHANGES_REQUESTED** [review]

File: erpnext/hr/doctype/appraisal/appraisal.js (hunk 1 above)
Comment (abhishekbalam) [reply erpnext/hr/doctype/appraisal/appraisal.js]: done

File: erpnext/hr/doctype/appraisal/appraisal.js [hunk 2]
```diff
@@ -10,45 +10,70 @@ frappe.ui.form.on('Appraisal', {
...
+			frm.set_value('total_score', 0);
+			return;
+		}
+		for (let i = 0; i<goals.length; i++) {
 			total = flt(total)+flt(goals[i].score_earned)
 		}
-		frm.set_value('total_score', total);
+		if (!isNaN(total)) {
+			frm.set_value('total_score', total);
+			frm.refresh_field('calculate_total');
```
Comment (Anurag810) [inline erpnext/hr/doctype/appraisal/appraisal.js:41]: ```frm.set_value``` already refresh field

**Review by Anurag810 — CHANGES_REQUESTED** [review]

File: erpnext/hr/doctype/appraisal/appraisal.json [hunk 3]
```diff
@@ -1,775 +1,254 @@
...
+  "start_date",
+  "end_date",
+  "department",
+  "section_break0",
+  "goals",
+  "total_score",
+  "section_break1",
+  "remarks",
+  "other_details",
+  "company",
```
Comment (Anurag810) [inline erpnext/hr/doctype/appraisal/appraisal.json:25]: Move the company field at the top.

File: erpnext/hr/doctype/appraisal/appraisal.json [hunk 4]
```diff
@@ -1,775 +1,254 @@
...
+  "column_break_17",
+  "amended_from"
```
Comment (Anurag810) [inline erpnext/hr/doctype/appraisal/appraisal.json:27]: Move this field to top tooo.

File: erpnext/hr/doctype/appraisal/appraisal.json (hunk 3 above)
Comment (Anurag810) [inline erpnext/hr/doctype/appraisal/appraisal.json:25]: The value of the company should be fetched from employee master. Keep the field read-only doesn't allow to change.
image

**Review by Anurag810 — CHANGES_REQUESTED** [review]

**Comment by abhishekbalam** [comment]
@Anurag810 
Changes made and gif updated.

**Comment by dawoodjee** [comment]
Why cant this be automated on change of the score field?

**Pull request merged**